### PR TITLE
support dup http resp headers

### PR DESCRIPTION
**Pull request by rentallect**

### CODE REVIEW

**Comment by ekoby** [comment]
replacing of the header via `set_` is needed for request headers, so that individual request can override client/default headers

I would prefer a separate function to add a header and used it for responses. we can even call it `add_http_header`

**Comment by rentallect** [comment]
I modified `set_http_headern` not `set_http_header`.  The `set_http_headern` func is only reachable during response parsing, and `set_http_header` is only used during request writing, so we should be fine, right?

**Comment by ekoby** [comment]
> I modified `set_http_headern` not `set_http_header`. The `set_http_headern` func is only reachable during response parsing, and `set_http_header` is only used during request writing, so we should be fine, right?

maybe it works, but I don't want `set_http_headern` and `set_http_header` to have different semantics. can you rename the changed function to `add_...`?

File: src/http_req.c [hunk 1]
```diff
@@ -117,7 +117,7 @@ size_t http_req_write(um_http_req_t *req, char *buf, size_t maxlen) {
     return len;
 }
 
-void set_http_headern(um_header_list *hl, const char* name, const char *value, size_t vallen) {
+void add_http_header(um_header_list *hl, const char* name, const char *value, size_t vallen) {
     um_http_hdr *h;
     LIST_FOREACH(h, hl, _next) {
```
Comment (ekoby) [inline src/http_req.c:122]: remove this search. not needed for add

Comment (rentallect) [reply src/http_req.c]: done

**Pull request merged**